dotCMS/src/main/java/com/dotmarketing/portlets/cmsmaintenance/ajax/CMSMaintenanceAjax.java

```diff
@@ -376,4 +428,293 @@ public void zipTempDirectoryToStream(OutputStream out) throws IOException {
+					//ordering is done only on the first field for the tables, which is INODE
+					if(com.dotmarketing.beans.Tree.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by parent, child, relation_type");
+					}
+					else if(MultiTree.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by parent1, parent2, child, relation_type");
+					}
+					else if(TagInode.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by inode, tag_id");
+					}
+					else if(TemplateVersionInfo.class.equals(clazz)){
+						_dh.setSQLQuery("SELECT {template_version_info.*} from template_version_info template_version_info, identifier where identifier.id = template_version_info.identifier order by template_version_info.identifier ");
+					}
+					else if(ContainerVersionInfo.class.equals(clazz)){
+						_dh.setSQLQuery("SELECT {container_version_info.*} from container_version_info container_version_info, identifier where identifier.id = container_version_info.identifier order by container_version_info.identifier ");
+					}
+					else if(LinkVersionInfo.class.equals(clazz)){
+						_dh.setSQLQuery("SELECT {link_version_info.*} from link_version_info link_version_info, identifier where identifier.id = link_version_info.identifier order by link_version_info.identifier ");
+					}
+					else if(CalendarReminder.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by user_id, event_id, send_date");
+					} else if(Identifier.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by parent_path, id");
+					} else {
+						_dh.setQuery("from " + clazz.getName() + " order by 1");
+
+					}
+
+					_list = _dh.list();
+					if(_list.size() ==0){
+						try {
+							_bout.close();
+						}
+						catch( java.lang.NullPointerException npe){}
+						_list = null;
+						_dh = null;
+						_bout = null;
+
+						break;
+					}
+
+					if(_list != null && _list.size() > 0 && _list.get(0) instanceof Comparable){
+						java.util.Collections.sort(_list);
+					}
+
+					_writing = new File(backupTempFilePath + File.separator +  clazz.getName() + "_" + formatter.format(i) + ".xml");
+					_bout = new BufferedOutputStream(Files.newOutputStream(_writing.toPath()));
+
+					total = total + _list.size();
+
+					try {
+						Thread.sleep(10);
+					} catch (InterruptedException e) {
+						Logger.warn(this, "An error ocurred trying to create XML files");
+						Logger.error(this,e.getMessage(),e);
+					}
+
+					try {
+						_xstream.toXML(_list, _bout);
+					} finally {
+						CloseUtils.closeQuietly(_bout);
+					}
+
+					_list = null;
+					_dh = null;
+					_bout = null;
+
+				}
+				Logger.info(this, "writing : " + total + " records for " + clazz.getName());
+			}
+
+			/* Run Liferay's Tables */
+			/* Companies */
+			_list = PublicCompanyFactory.getCompanies();
+			List<Company> companies = new ArrayList<Company>(_list);

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "companies" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)